### PR TITLE
Add form action

### PR DESCRIPTION
**Pull request by AndrewKorzh**
# Pull Request: Add FillForm Class for Form Automation
# Overview
This pull request introduces a new class FillForm, designed to automate the process of filling out and submitting forms on a webpage. The FillForm class allows users to specify input fields with their corresponding values and optional delays, as well as interact with a submit button if provided.

# New Feature
Class: FillForm
This class extends PuppeteerServiceAction and is used to fill out forms based on CSS selectors. It also handles optional delays between keystrokes and submits the form if a submit button is specified.
Class Definition
The FillForm class is defined as follows:

```python
class FillForm(PuppeteerServiceAction):
    """
    Fill out and submit forms on a webpage.

    Available options:

    * ``input_mapping`` (dict): A dictionary where each key is a CSS selector, and
    each value is another dictionary containing details about the input for that element.
    Each entry in the dictionary should follow this structure:

    * ``selector`` (str): The CSS selector for the input element (used as the key).
    * ``value`` (str): The text to be inputted into the element.
    * ``delay`` (int, optional): A delay (in milliseconds) between each keystroke
        when inputting the text. Defaults to 0 if not provided.

    * ``submit_button`` (str, optional): The CSS selector for the form's submit button.
    If provided, the button will be clicked after filling in the form.
    """

    endpoint = "fill_form"

    def __init__(self, input_mapping: dict, submit_button: str = None):
        self.input_mapping = input_mapping
        self.submit_button = submit_button

    def payload(self):
        return {"inputMapping": self.input_mapping, "submitButton": self.submit_button}
```

# Key Points
input_mapping: A dictionary that maps CSS selectors to input details. Each selector is paired with a dictionary containing:

value: The text to be entered into the input element.
delay (optional): Delay in milliseconds between keystrokes.
submit_button (optional): CSS selector for the submit button, which will be clicked after the form is filled.

# Example Usage
To use the FillForm class, initialize it with the required input_mapping and an optional submit_button:

``` python
form_action = FillForm(
    input_mapping={
        "#username": {"value": "exampleUser", "delay": 100},
        "#password": {"value": "examplePass", "delay": 100}
    },
    submit_button="#loginButton"
)
```
This configuration will fill in the username and password fields with the provided values and delays, and click the submit button with the ID #loginButton.

### CODE REVIEW

File: examples/spiders/form_action.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import scrapy
+from scrapypuppeteer import PuppeteerRequest, PuppeteerScreenshotResponse
+from scrapypuppeteer.actions import Screenshot, FormAction
+import base64
+
+class FormActionSpider(scrapy.Spider):
+    name = 'form_action'
```
Comment (MatthewZMSU) [inline examples/spiders/form_action.py]: Do not ignore PEP8. Imports should be sorted. Class implementations should be 2 indentations apart from imports.

File: examples/spiders/har.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+from scrapypuppeteer import PuppeteerRequest
+from scrapypuppeteer.actions import Har
+
+
+def write_to_file(file_path, content):
+    with open(file_path, 'a', encoding='utf-8') as file:
+        file.write(content)
+
+class HarSpider(scrapy.Spider):
+    name = 'har'
```
Comment (MatthewZMSU) [inline examples/spiders/har.py]: PEP8

File: scrapypuppeteer/actions.py [hunk 3]
```diff
@@ -226,10 +226,51 @@ def payload(self):
...
+    you must pass the `har_recording=True` argument to `PuppeteerRequest`
+    when initiating the request.
+    """
     endpoint = "har"
 
     def payload(self):
         return {}
+    
+    
+class FormAction(PuppeteerServiceAction):
```
Comment (MatthewZMSU) [inline scrapypuppeteer/actions.py]: Maybe the action should be called like `FillForm` or `CompleteForm`. As we know that this class performs action.

File: examples/spiders/form_action.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    start_urls = ["https://www.roboform.com/filling-test-all-fields"]
+
+    def start_requests(self):
```
Comment (MatthewZMSU) [inline examples/spiders/form_action.py]: When started with `PUPPETEER_LOCAL` it gave many extra lines in the logs.
Should be fixed in other PR.

**Review by MatthewZMSU — DISMISSED** [review]
LGTM. Good simple logic.

Please:
1.) Do not ignore Pull Request template. Fill it please so that the reviewer would see the changes and motivation. Look for other PRs in hostory for examples.
2.) Do not ignore PEP8. It is one of the main aspects of Python. Use `black` for formatting.

File: examples/spiders/har.py (hunk 2 above)
Comment (AndrewKorzh) [reply examples/spiders/har.py]: Done! :)

File: examples/spiders/form_action.py (hunk 1 above)
Comment (AndrewKorzh) [reply examples/spiders/form_action.py]: Done! :)

File: examples/spiders/form_action.py (hunk 4 above)
Comment (AndrewKorzh) [reply examples/spiders/form_action.py]: I don't really understand where this problem is coming from...

**Review by MatthewZMSU — DISMISSED** [review]

**Review by MatthewZMSU — APPROVED** [review]

**Pull request merged**